lava/classes/player.py

```diff
@@ -74,26 +75,31 @@ async def check_autoplay(self) -> bool:
 
         :return: True if tracks were added, False otherwise.
         """
-        if not self.autoplay or len(self.queue) >= 5:
-            return False
-        self.bot.logger.info(
-            "Queue is empty, adding recommended track for guild %s...", self.guild
-        )
+        while True:
+            if not self.autoplay or len(self.queue) >= 5:
+                await asyncio.sleep(1)
+                continue
+
+            self.bot.logger.info(
+                "Queue is empty, adding recommended track for guild %s...", self.guild
+            )
 
-        recommendations = await get_recommended_tracks(self, self.current, 5 - len(self.queue))
+            recommendations = await get_recommended_tracks(self, self.current, 5 - len(self.queue))
 
-        for recommendation in recommendations:
-            self.add(requester=0, track=recommendation)
+            for recommendation in recommendations:
+                self.add(requester=0, track=recommendation)
 
     async def toggle_autoplay(self):
         """
         Toggle autoplay for the player.
         """
         if not self.autoplay:
             self.autoplay = True
+            self.autoplay_task = self.bot.loop.create_task(self.check_autoplay())
             return
 
         self.autoplay = False
+        self.autoplay_task.cancel()
```

Comment:
I haven't dig the code deep. But can this make sure that the task is canceled when the player is destroyed?